src/types.rs

```diff
@@ -248,12 +248,22 @@ fn encode_prefix_hex(bytes: &[u8]) -> String {
     format!("0x{}", faster_hex::hex_string(bytes))
 }
 
+fn map_address_from_binary(v: &Option<FixedSizeData<20>>, should_checksum: bool) -> Option<String> {
+    v.as_ref().map(|v| {
+        if should_checksum {
+            alloy_primitives::Address(alloy_primitives::FixedBytes(***v)).to_checksum(None)
+        } else {
+            v.encode_hex()
+        }
+    })
+}
+
 fn map_binary<T: Hex>(v: &Option<T>) -> Option<String> {
     v.as_ref().map(|v| v.encode_hex())
 }
 
-impl From<&simple_types::Block> for Block {
-    fn from(b: &simple_types::Block) -> Self {
+impl Block {
+    pub fn from_simple(b: &simple_types::Block, should_checksum: bool) -> Self {
```

Comment:
Remove the trait implementation and add method called from_simple in order to pass flag as an argument. Added the `_simple` suffix to avoid any confusion with the "from" trait.